### PR TITLE
Offline-KD QAD example

### PR DESCRIPTION
**Pull request by AAnoosheh**
### What does this PR do?

Type of change: new feature

Adds a Nano-v3 launcher example which uses the new MLM offline-logits KD feature

### Usage

```python
# Add a code snippet demonstrating how to use this
```

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, `torch.load(..., weights_only=False)`, `pickle`, etc.).

- Is this change backward compatible?: N/A
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: N/A
- Did you write any new necessary tests?: N/A
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: ❌ 
- Did you get Claude approval on this PR?: ❌ 

## Summary by CodeRabbit

* **New Features**
  * Added a new offline, five-stage Slurm workflow example for Nemotron-3-Nano 30B NVFP4 A3B, using cached teacher logits for knowledge distillation.
  * Runs quantization, frozen teacher logits generation, student fine-tuning from saved logits, checkpoint export to Hugging Face format, and TensorRT-LLM evaluation.
  * Includes ready-to-run configuration for repeatable execution with tuned parallelism and export settings.
* **Chores**
  * Updated local Python pre-commit hooks to execute via `uv` for consistent dev-time tooling.

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (kevalmorabia97) [inline .pre-commit-config.yaml:1]: what do these changes do? Is it supposed to make it faster?

Comment (AAnoosheh) [reply .pre-commit-config.yaml:1]: I got a new machine and decided during setup I will avoid system python and just use `uv` for project dependencies. That way I don't need to worry about installing the pre-commit deps like `pyyaml` into something outside of this project's `uv`.

File: tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    slurm_config:
+      _factory_: "slurm_factory"
+      container: nvcr.io/nvidia/nemo:26.06
+      modelopt_install_path: /opt/venv/lib/python3.12/site-packages/modelopt
+      nodes: 1
+      ntasks_per_node: 4
+      gpus_per_node: 4
+
+  task_1:
+    script: common/megatron_lm/train/sft.sh
```
Comment (ChenhanYu) [inline tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml:67]: Ideally we use the shell script in megatron-lm directly. I think it is possible but require refactorization of the script.

File: tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      - --adam-beta2 0.95
+      - --init-method-std 0.010
+      - --use-distributed-optimizer
+      - --eval-iters 4
+      - --eval-interval 1000
+      - --save-interval 1000
+      - --log-interval 100
+      - --finetune
+      - --freeze-all-layers
+      - --logits-save-top-k 16
```
Comment (ChenhanYu) [inline tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml:104]: So these 2 args to save the logits?

**Comment by ChenhanYu** [comment]
So 2 args to save and 1 args to enable offline kd?

File: tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml (hunk 2 above)
Comment (AAnoosheh) [reply tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml:104]: Yes. (Soon technically it will be only 1 arg since in next update I'll make default k=128)

File: tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml (hunk 1 above)
Comment (AAnoosheh) [reply tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml:67]: wdym?

**Review by ChenhanYu — APPROVED** [review]

File: tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+#   uv run launch.py --yaml examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml --yes
+
+job_name: NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4-QAD-OfflineKD
+pipeline:
+  task_0:
+    script: common/megatron_lm/quantize/quantize.sh
+    args:
+      # In-domain calibration on the SFT corpus (raw .jsonl, read via the calib
+      # loader's "messages" branch) instead of cnn_dailymail -- better-matched amax.
+      - --calib-dataset-path-or-name /hf-local/modelopt/nemotron-3-super-sft/train.jsonl
```
Comment (kevalmorabia97) [inline tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml:46]: Is this something external users can also download and use?

File: tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+      - --async-save
+      - --use-persistent-ckpt-worker
+    environment:
+      - CP: 8
+      - TP: 1
+      - PP: 1
+      - EP: 4
+      - ETP: 1
+      - MLM_MODEL_CFG: nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16
+      - MLM_MODEL_CKPT: /hf-local/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16-MCore
```
[the remark this answers is not in the document]
Comment (kevalmorabia97) [reply tools/launcher/examples/nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-NVFP4/offline_kd_qad.yaml:115]: @AAnoosheh perhaps we can add a note on how this is produced? Or do it on the fly with Megatron-Bridge's convert_checkpoints.py file?

File: .pre-commit-config.yaml
Comment (kevalmorabia97) [reply .pre-commit-config.yaml:1]: sounds good
